function/src/main/java/com/example/driver/DemoApplication.java

```diff
@@ -2,82 +2,68 @@
 
 import java.util.function.Function;
 
-import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.builder.CloudEventBuilder;
+import io.cloudevents.core.provider.EventFormatProvider;
+import io.cloudevents.protobuf.ProtobufFormat;
+import io.cloudevents.spring.messaging.CloudEventMessageConverter;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 import org.springframework.core.io.ClassPathResource;
-import org.springframework.messaging.Message;
-import org.springframework.messaging.MessageHeaders;
-import org.springframework.messaging.converter.MessageConverter;
-import org.springframework.messaging.support.MessageBuilder;
-import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
 import org.springframework.wasm.WasmLoader;
 import org.springframework.wasm.WasmRunner;
 
 @SpringBootApplication
 class DemoApplication {
 
-	public static void main(String[] args) throws Exception {
+	public static void main(String[] args) {
 		SpringApplication.run(DemoApplication.class, args);
 	}
 
-}
-
-@Component
-class WASMFunction implements AutoCloseable, Function<SpringMessage, SpringMessage> {
-
-	private WasmLoader wasmLoader = new WasmLoader();
-
-	@Override
-	public void close() throws Exception {
-		this.wasmLoader.close();
-	}
-
-	@Override
-	public SpringMessage apply(SpringMessage input) {
-		try (WasmRunner runner = wasmLoader.runner(new ClassPathResource("message.wasm"))) {
-			return runner == null ? input : runner.call("filter", input, SpringMessage.class);
-		}
-	}
-
-}
-
-@Component
-class SpringMessageConverter implements MessageConverter {
+	@Bean
+	Function<CloudEvent, CloudEvent> decorateEventWithWasm(WasmLoader wasmLoader) {
+		return (input) -> {
+			// Convert the incoming CE to proto CE
+			io.cloudevents.v1.proto.CloudEvent inputProto;
+			try {
+				byte[] inputProtoBytes = new ProtobufFormat().serialize(input);
+				inputProto = io.cloudevents.v1.proto.CloudEvent.parseFrom(inputProtoBytes);
```

Comment:
There may be a more direct, efficient way to go from `CloudEvent` object to `CloudEvent` proto object but was spending time tweaking on that so, again, in the spirit of forward progress went w/ approach that worked.